test/versioned/bluebird/helpers.js

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+const assert = require('node:assert')
+const { runMultiple } = require('../../lib/promises/helpers')
+const { tspl } = require('@matteo.collina/tspl')
+const symbols = require('../../../lib/symbols')
+const helper = require('../../lib/agent_helper')
+const util = require('util')
+const setImmediatePromisified = util.promisify(setImmediate)
+
+async function beforeEach(ctx) {
+  ctx.nr = {}
+  ctx.nr.agent = helper.instrumentMockedAgent()
+  ctx.nr.Promise = require('bluebird')
+  ctx.nr.tasks = []
+  ctx.nr.interval = setInterval(function () {
+    if (ctx.nr.tasks.length) {
+      ctx.nr.tasks.pop()()
+    }
+  }, 25)
+
+  await setImmediatePromisified()
+}
+
+async function afterEach(ctx) {
+  helper.unloadAgent(ctx.nr.agent)
+  clearInterval(ctx.nr.interval)
+
+  await setImmediatePromisified()
+}
+
+function id(tx) {
+  return tx && tx.id
+}
+
+function addTask() {
+  const args = [].slice.apply(arguments)
+  const { tasks } = args.shift() // Pop test context
+  const fn = args.shift() // Pop function from args
+  tasks.push(function () {
+    fn.apply(null, args)
+  })
+}
+
+function testPromiseClassMethod({ t, count, testFunc, end }) {
+  testPromiseMethod({ t, count, factory: testFunc, end })
+}
+
+function testPromiseInstanceMethod({ t, count, testFunc, end }) {
+  const { Promise } = t.nr
+  testPromiseMethod({
+    t,
+    end,
+    count,
+    factory: function ({ plan, name }) {
+      const promise = Promise.resolve([1, 2, 3, name])
+      return testFunc({ plan, name, promise })
+    }
+  })
+}
+
+function testPromiseMethod({ t, count, factory, end }) {
+  const { agent } = t.nr
+  const COUNT = 2
+  const plan = tspl(t, { plan: count * 3 + (COUNT + 1) * 3 })
+
+  plan.doesNotThrow(function outTXPromiseThrowTest() {
+    const name = '[no tx] '
+    let isAsync = false
+    factory({ plan, name })
+      .finally(function () {
+        plan.ok(isAsync, name + 'should have executed asynchronously')
+      })
+      .then(
+        function () {
+          plan.ok(!agent.getTransaction(), name + 'has no transaction')
+          testInTransaction()
+        },
+        function (err) {
+          if (err) {
+            /* eslint-disable no-console */
+            console.log(err.stack)
```

Comment:
Will this, and the other `console.log`s in this file, literally write to stdout during test runs? Or will they write to a trapped stream?

Reply:
hmm, not sure. let me dig, this is just code that was ported